Mathlib/LinearAlgebra/Finrank.lean

```diff
@@ -65,6 +65,10 @@ theorem finrank_eq_of_rank_eq {n : ℕ} (h : Module.rank K V = ↑n) : finrank K
   exact_mod_cast h
 #align finite_dimensional.finrank_eq_of_rank_eq FiniteDimensional.finrank_eq_of_rank_eq
 
+lemma rank_eq_one_iff_finrank_eq_one :
+    Module.rank K V = 1 ↔ FiniteDimensional.finrank K V = 1 := by
+  simp [FiniteDimensional.finrank]
```

Comment:
I don't yet understand `ofNat`; is this a syntactic generalization?
```suggestion
lemma rank_eq_ofNat_iff_finrank_eq_ofNat :
    Module.rank K V = ofNat n ↔ FiniteDimensional.finrank K V = ofNat n := by
  simp [FiniteDimensional.finrank] -- probably missing a simp lemma
```

Reply:
No, it doesn't work because there is no `OfNat.ofNat n : Cardinal` for a general natural `n`; only for concrete numbers such as `0`, `1`, `2`,...

Reply:
This works:
```
lemma rank_eq_ofNat_iff_finrank_eq_ofNat (n : ℕ) [Nat.AtLeastTwo n] :
    Module.rank K V = OfNat.ofNat n ↔ FiniteDimensional.finrank K V = OfNat.ofNat n := by
  simp [FiniteDimensional.finrank]
```
Was I meant to read your message as `0`, `1`, or `2,3,4,5,...`?

Reply:
Either way, that can be in a separate PR.

Reply:
I'm investigating. There is something very unpleasant currently, that for any concrete integer `n` (say `0`, or `3`, or `17`), we have `(37 : Cardinal) = ((37 : ℕ) : Cardinal)`, except for the integer `1`. This is due to the current definition of `natCast`.

Reply:
#6362 has the lemma above

Reply:
I have merged #6362 (and removed the duplicate lemma from the current PR).